### DISCUSSION TITLE
Problems flashing Tasmota on Wemos pro

### DISCUSSION THREAD

**Discussion by hunmik** [arendst/Tasmota#22692] [Q&A]
**device **
I have some wemos D1 mini pro devices. Main difference from other wemos that they have 16M memory, and a CP2014 USB-serial chip.
Some details here: https://www.monohelixlabs.com/wemos-d1-mini-pro-esp8266.html
 A HTU21 sensor  is connected through SCL-TX SDA-RX lines. The firmware is the pre-compiled tasmota-sensors.bin

**problem**
Problem is that I cannot flash Tasmota on them or if I can, Tasmota gets corrupted after some time of operation. 
As this is a normal ESP8266, I see no reason why.

**environment**
For flashing I use a laptop with windows 10, regular usb cable.
With Arduino I can flash any code (other than Tasmota) on this devices and they work well. On the same computer, same driver same cable, same port etc. so I do not see any possible problem with those things.

**with online tool**
With online install tool (https://tasmota.github.io/install/) after selection the device and starting flashing I get the error message: "Failed to initialize. Try resetting your device or holding the BOOT button while clicking INSTALL.". There is no BOOT button, but even if it had it should work without that.
I can flash Tasmota to other ESP devices with the online tool without problem.

**with esptool**
If I flash with GIO, it completes successfully, but Tasmota **will not run**:
PS C:\work\esp> .\esptool.exe --port COM5 write_flash -fs 4MB -fm qio 0x0 tasmota-sensors.bin

If I flash with DOUT, it completes successfully.
PS C:\work\esp>  .\esptool.exe --port COM5 write_flash -fs 4MB -fm dout 0x0 tasmota-sensors.bin
**In this way Tasmota runs fine, but after some days of operation memory gets corrupted, device name changes to garbage, and settings also change to invalid values**

I'm not familiar with Tasmota (yet) so any suggestion is highly apprecrated.

**Comment by sfromis** [comment]
Well, flash mode DOUT is the right one, as indicated by Tasmota running fine after flashing, meaning that there were no problems flashing. I usually prefer flashing with `--flash_size detect` letting it select the right size, and `--erase-all` to completely erase flash.

The corruption you see "after some days", is where you have an issue, which might very well be hardware-related. Tasmota is likely much larger than the code you flash from Arduino, which can make a difference if the flash chip is dubious. Ensuring good power while flashing may also be important, like not going through an USB hub. Flash chip from Puya used to have issues, not sure what the state is here.

Using TX/RX for I2C may not be the best plan, might disturb serial during flashing, and with the USB/serial adapter chip "owning" the pins I'd select regular free pins, unless in a very tight spot.

I've had zero problems with a couple of "Wemos D1 mini Pro" boards.

**Reply by hunmik** [reply]
Thank you for the hints. As first step I have moved my IO connections from TX/RX and reflashed. So far it works well, but I need to test it for some weeks before I can say "problem solved".
It is good to know, that there is no general problem with WemosD1 mini Pro.

**Reply by hunmik** [reply]
Now after 5 days of operation, device stopped sending MQTT. I'have checked, and the issue is that in the topic the %pr**e**fix% has changed to %pr**a**fix%. I have no idea how it could happen. I have corrected it and testing again.
